uniffi_bindgen/src/interface/function.rs

```diff
@@ -184,10 +188,12 @@ impl APIConverter<Argument> for weedle::argument::SingleArgument<'_> {
             Some(v) => Some(convert_default_value(&v.value, &type_)?),
         };
         let by_ref = ArgumentAttributes::try_from(self.attributes.as_ref())?.by_ref();
+        let by_arc = ArgumentAttributes::try_from(self.attributes.as_ref())?.by_arc();
```

Comment:
I'm actually wondering if `[ByArc]` is really necessary on arguments, given that we've simplified the proposal down quite a bit. If you're passing something that's not an `Object` instance then `[ByArc]` is invalid. If you're passing something that is an `Object` instance then you *have* to be passing it in an `Arc`. So I'm not sure what value the annotation provides, except maybe for reminding the component author that they need to accept an `Arc<T>` instead of a `T`. (But even then, the Rust compiler will also remind them of this fact).

Reply:
> If you're passing something that is an Object instance then you have to be passing it in an Arc.

Hrm, I guess this isn't necessarily true - the bindings could take an `&T` out of the `Arc<T>` and pass it to the hand-written Rust code. But we could consider having `f(T arg)` mean "receives an owned value in an Arc" and `f([ByRef] T arg)` mean "receives a borrowed reference", which is how it works for non-Object types.